src/twisted/conch/ssh/session.py

```diff
@@ -91,6 +92,33 @@ def request_pty_req(self, data):
         else:
             return 1
 
+    def request_env(self, data):
+        """
+        Process a request to pass an environment variable.
+
+        @param data: The environment variable name and value, each encoded
+            as an SSH protocol string and concatenated.
+        @type data: L{bytes}
+        @return: A true value if the request to pass this environment
+            variable was accepted, otherwise a false value.
+        """
+        if not self.sessionSetEnv:
```

Comment:
This seems like a good candidate for the "active nothing" pattern, where it's never `None` but sometimes it's a specialized no-op implementation that refuses all variables?  This pushes some of the conditional logic below into the stub implementation, making it less complex.

Reply:
I'd prefer not to do this in this case, because that loses the distinction for logging purposes between "can't set environment variables because doing that isn't implemented" and "can't set environment variables because the implementation said no", which seems to me to be likely to be useful.  (Or, if I were to preserve the extra logging, as far as I can see it would make no difference to the complexity of the conditional logic below it: it still needs to cope with the three cases of OK, explicitly-forbidden, and error.)